src/api/local-storage-provider.model.ts

```diff
@@ -2,67 +2,136 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import PouchDB from "pouchdb-browser";
-import { isScheduleModelEmpty, ScheduleDataModel } from "../common-models/schedule-data.model";
-import { ScheduleDataActionCreator } from "../state/reducers/month-state/schedule-data/schedule-data.action-creator";
 import {
-  PersistanceStoreProvider,
+  cropScheduleDMToMonthDM,
+  getScheduleKey,
+  isMonthModelEmpty,
+  MonthDataModel,
+  ScheduleDataModel,
+} from "../common-models/schedule-data.model";
+
+import {
+  MonthRevision,
+  PersistenceStoreProvider,
   RevisionFilter,
   RevisionType,
-  ScheduleRevision,
-  ThunkFunction,
+  ScheduleKey,
+  UpdatePosition,
 } from "./persistance-store.model";
+import _ from "lodash";
+import { calculateMissingFullWeekDays } from "../state/reducers/month-state/schedule-data/common-reducers";
+import { ShiftInfoModel } from "../common-models/shift-info.model";
 
 export const DATABASE_NAME = "nurse-scheduling";
-/*eslint-disable @typescript-eslint/camelcase */
-export class LocalStorageProvider extends PersistanceStoreProvider {
-  private storage: PouchDB.Database<ScheduleRevision>;
+
+export class LocalStorageProvider extends PersistenceStoreProvider {
+  private storage: PouchDB.Database<MonthRevision>;
 
   constructor() {
     super();
     this.storage = new PouchDB(DATABASE_NAME);
   }
 
-  saveScheduleRevision(
+  async saveMonthRevision(type: RevisionType, monthDataModel: MonthDataModel): Promise<void> {
```

Comment:
W bazie przechowujemy tylko miesiące, więc jeśli przekazujemy do zapisuje MonthDataModel to schemat jest prosty